### ISSUE TITLE
HDF5 errors

### ISSUE THREAD

**Issue by JuliusSustarevas** [ros-industrial-consortium/reuleaux#60]
Hi, sorry if this is a silly question but:
I am building a reachablity map for  the panda arm.  ubuntu 16, kinetic
**I've set everything up and when actually creating the map I get a whole bunch of hdf5 errors.:**

amlnuc@amlnuc2:~/catkin_ws$ rosrun map_creator create_reachability_map 0.08 panda.h5
[ INFO] [1582216936.895534878]: Creating the box and discretizing with resolution: 0.080000
[ INFO] [1582216936.940729633]: Total no of spheres now: 54872
[ INFO] [1582216936.940780666]: Please hold ON. Spheres are discretized and all of the poses are checked for Ik solutions. May take some time
[ INFO] [1582217461.845437242]: Total number of poses: 2743600
[ INFO] [1582217461.845489157]: Total number of reachable poses: 0
[ INFO] [1582217461.845501440]: No of spheres reachable: 0
[ INFO] [1582217461.845689909]: Creating Directory
[ INFO] [1582217461.845721518]: Saving map panda.h5
[ INFO] [1582217461.860297280]: Saving poses in reachability map
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5Pdcpl.c line 870 in H5Pset_chunk(): all chunk dimensions must be positive
    major: Invalid arguments to routine
    minor: Out of range
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5D.c line 194 in H5Dcreate2(): unable to create dataset
    major: Dataset
    minor: Unable to initialize object
  #001: ../../../src/H5Dint.c line 453 in H5D__create_named(): unable to create and link to dataset
    major: Dataset
    minor: Unable to initialize object
  #002: ../../../src/H5L.c line 1638 in H5L_link_object(): unable to create new link to object
    major: Links
    minor: Unable to initialize object
  #003: ../../../src/H5L.c line 1882 in H5L_create_real(): can't insert link
    major: Symbol table
    minor: Unable to insert object
  #004: ../../../src/H5Gtraverse.c line 861 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #005: ../../../src/H5Gtraverse.c line 641 in H5G_traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #006: ../../../src/H5L.c line 1685 in H5L_link_cb(): unable to create object
    major: Object header
    minor: Unable to initialize object
  #007: ../../../src/H5O.c line 3016 in H5O_obj_create(): unable to open object
    major: Object header
    minor: Can't open object
  #008: ../../../src/H5Doh.c line 293 in H5O__dset_create(): unable to create dataset
    major: Dataset
    minor: Unable to initialize object
  #009: ../../../src/H5Dint.c line 1056 in H5D__create(): unable to construct layout information
    major: Dataset
    minor: Unable to initialize object
  #010: ../../../src/H5Dchunk.c line 516 in H5D__chunk_construct(): no chunk information set?
    major: Dataset
    minor: Bad value
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5D.c line 993 in H5Dset_extent(): not a dataset
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5D.c line 460 in H5Dget_space(): not a dataset
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5Shyper.c line 6593 in H5Sselect_hyperslab(): not a data space
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5Dio.c line 228 in H5Dwrite(): not a dataset
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5S.c line 392 in H5Sclose(): not a dataspace
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5D.c line 993 in H5Dset_extent(): not a dataset
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5D.c line 460 in H5Dget_space(): not a dataset
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5Shyper.c line 6593 in H5Sselect_hyperslab(): not a data space
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5Dio.c line 228 in H5Dwrite(): not a dataset
    major: Invalid arguments to routine
    minor: Inappropriate type
[ INFO] [1582217461.860760272]: Saving spheres in Reachability map
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5S.c line 392 in H5Sclose(): not a dataspace
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140411279792128:
  #000: ../../../src/H5D.c line 415 in H5Dclose(): not a dataset
    major: Invalid arguments to routine
    minor: Inappropriate type
[ INFO] [1582217461.860976191]: Elasped time is 525.00 seconds.
[ INFO] [1582217461.860991632]: Completed
`

**The file is still created. However then when I try to load it, it fails:**

amlnuc@amlnuc2:~/catkin_ws/src/my_reuleaux_robots/map_creator/maps$ rosrun map_creator load_reachability_map panda.h5
[ INFO] [1582218196.756448321]: Opening map panda.h5
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140261651089280:
  #000: ../../../src/H5D.c line 358 in H5Dopen2(): not found
    major: Dataset
    minor: Object not found
  #001: ../../../src/H5Gloc.c line 430 in H5G_loc_find(): can't find object
    major: Symbol table
    minor: Object not found
  #002: ../../../src/H5Gtraverse.c line 861 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #003: ../../../src/H5Gtraverse.c line 641 in H5G_traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #004: ../../../src/H5Gloc.c line 385 in H5G_loc_find_cb(): object 'poses_dataset' doesn't exist
    major: Symbol table
    minor: Object not found
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140261651089280:
  #000: ../../../src/H5D.c line 460 in H5Dget_space(): not a dataset
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140261651089280:
  #000: ../../../src/H5S.c line 791 in H5Sget_simple_extent_ndims(): not a dataspace
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.8.16) thread 140261651089280:
  #000: ../../../src/H5S.c line 883 in H5Sget_simple_extent_dims(): not a dataspace
    major: Invalid arguments to routine
    minor: Inappropriate type
Bus error (core dumped)

**Comment by ichdich** [comment]
[ INFO] [1582217461.845489157]: Total number of reachable poses: 0
[ INFO] [1582217461.845501440]: No of spheres reachable: 0
I think this may cause the hdf5 errors.
You may have incorrect ikfast61_xxx.cpp file.

**Comment by jontromanab** [comment]
Yes. I can agree with @ichdich . YOu have a wrong ikfast file which did not provide any IK solution. That is the reason there is no reachable poses. So the map generation process generated spheres, but no poses. 
The rest of the errors are just hdf5 errors as the system tried to push empty data and accessed those.
I have to make some safeguards in some PR. :)
